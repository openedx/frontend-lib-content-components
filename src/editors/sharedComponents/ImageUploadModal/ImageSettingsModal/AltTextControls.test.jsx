import React from 'react';
import { shallow } from '@edx/react-unit-test-utils';

import { formatMessage } from '../../../../testUtils';
import { AltTextControls } from './AltTextControls';

jest.mock('./hooks', () => ({
  onInputChange: (handler) => ({ 'hooks.onInputChange': handler }),
  onCheckboxChange: (handler) => ({ 'hooks.onCheckboxChange': handler }),
}));

describe('AltTextControls', () => {
  const props = {
    isDecorative: true,
    value: 'props.value',
    // inject
    intl: { formatMessage },
  };
  beforeEach(() => {
    props.setValue = jest.fn().mockName('props.setValue');
    props.setIsDecorative = jest.fn().mockName('props.setIsDecorative');
    props.validation = { show: true };
  });
  describe('render', () => {
    test('snapshot: isDecorative=true errorProps.showAltTextSubmissionError=true', () => {
      expect(shallow(<AltTextControls {...props} />).snapshot).toMatchSnapshot();
    });
    test('snapshot: isDecorative=true errorProps.showAltTextSubmissionError=false', () => {
      props.validation.show = false;
      expect(shallow(<AltTextControls {...props} />).snapshot).toMatchSnapshot();
    });
  });
});
