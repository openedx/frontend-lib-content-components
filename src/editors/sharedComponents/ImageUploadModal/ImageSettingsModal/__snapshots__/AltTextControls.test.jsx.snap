// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AltTextControls render snapshot: isDecorative=true errorProps.showAltTextSubmissionError=false 1`] = `
<Form.Group
  className="mt-4.5"
>
  <Form.Label
    as="h4"
  >
    <FormattedMessage
      defaultMessage="Accessibility"
      description="Label title for accessibility section."
      id="authoring.texteditor.imagesettingsmodal.accessibilityLabel"
    />
  </Form.Label>
  <Form.Control
    className="mt-4.5"
    disabled={true}
    floatingLabel="Alt Text"
    isInvalid={false}
    onChange={
      {
        "hooks.onInputChange": [MockFunction props.setValue],
      }
    }
    type="input"
    value="props.value"
  />
  <Form.Checkbox
    checked={true}
    className="mt-4.5 decorative-control-label"
    onChange={
      {
        "hooks.onCheckboxChange": [MockFunction props.setIsDecorative],
      }
    }
  >
    <Form.Label>
      <FormattedMessage
        defaultMessage="This image is decorative (no alt text required)."
        description="Checkbox label for whether or not an image is decorative."
        id="authoring.texteditor.imagesettingsmodal.decorativeAltTextCheckboxLabel"
      />
    </Form.Label>
  </Form.Checkbox>
</Form.Group>
`;

exports[`AltTextControls render snapshot: isDecorative=true errorProps.showAltTextSubmissionError=true 1`] = `
<Form.Group
  className="mt-4.5"
>
  <Form.Label
    as="h4"
  >
    <FormattedMessage
      defaultMessage="Accessibility"
      description="Label title for accessibility section."
      id="authoring.texteditor.imagesettingsmodal.accessibilityLabel"
    />
  </Form.Label>
  <Form.Control
    className="mt-4.5"
    disabled={true}
    floatingLabel="Alt Text"
    isInvalid={true}
    onChange={
      {
        "hooks.onInputChange": [MockFunction props.setValue],
      }
    }
    type="input"
    value="props.value"
  />
  <Form.Control.Feedback
    type="invalid"
  >
    <FormattedMessage
      defaultMessage="Enter alt text"
      description="Message feedback for user below the alt text field."
      id="authoring.texteditor.imagesettingsmodal.error.altTextLocalFeedback"
    />
  </Form.Control.Feedback>
  <Form.Checkbox
    checked={true}
    className="mt-4.5 decorative-control-label"
    onChange={
      {
        "hooks.onCheckboxChange": [MockFunction props.setIsDecorative],
      }
    }
  >
    <Form.Label>
      <FormattedMessage
        defaultMessage="This image is decorative (no alt text required)."
        description="Checkbox label for whether or not an image is decorative."
        id="authoring.texteditor.imagesettingsmodal.decorativeAltTextCheckboxLabel"
      />
    </Form.Label>
  </Form.Checkbox>
</Form.Group>
`;
